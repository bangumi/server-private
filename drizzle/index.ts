export * from './db.ts';
export * as orm from './orm.ts';
export * as schema from './schema.ts';
export * as op from 'drizzle-orm';
