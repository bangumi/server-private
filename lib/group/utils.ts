import { db, op, schema } from '@app/drizzle';

/** Is user member of group */
export async function isMemberInGroup(userID: number, groupID: number): Promise<boolean> {
  const [d] = await db
    .select({ uid: schema.chiiGroupMembers.uid, gid: schema.chiiGroupMembers.gid })
    .from(schema.chiiGroupMembers)
    .where(
      op.and(
        op.eq(schema.chiiGroupMembers.uid, userID),
        op.eq(schema.chiiGroupMembers.gid, groupID),
      ),
    )
    .limit(1);

  return Boolean(d);
}
