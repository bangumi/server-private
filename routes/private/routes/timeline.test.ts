import { afterEach, beforeEach, describe, expect, test } from 'vitest';

import { db, op, schema } from '@app/drizzle';
import { emptyAuth } from '@app/lib/auth/index.ts';
import { createTestServer } from '@app/tests/utils.ts';

import { setup } from './timeline.ts';

describe('timeline', () => {
  test('should get friends', async () => {
    const app = createTestServer({
      auth: {
        ...emptyAuth(),
        login: true,
        userID: 287622,
      },
    });

    await app.register(setup);
    const res = await app.inject({
      method: 'get',
      url: '/timeline',
      query: { mode: 'friends' },
    });
    expect(res.json()).toMatchSnapshot();
  });
});

describe('timeline status', () => {
  beforeEach(async () => {
    await db
      .delete(schema.chiiTimeline)
      .where(op.and(op.eq(schema.chiiTimeline.uid, 287622), op.eq(schema.chiiTimeline.cat, 5)));
  });

  afterEach(async () => {
    await db
      .delete(schema.chiiTimeline)
      .where(op.and(op.eq(schema.chiiTimeline.uid, 287622), op.eq(schema.chiiTimeline.cat, 5)));
  });

  test('should create timeline say', async () => {
    const app = createTestServer({
      auth: {
        ...emptyAuth(),
        login: true,
        userID: 287622,
      },
    });

    await app.register(setup);
    const res = await app.inject({
      method: 'post',
      url: '/timeline',
      body: { content: '^_^\n(bgm38)>_<', turnstileToken: 'fake' },
    });
    expect(res.statusCode).toBe(200);
    const id = res.json().id;

    const [data] = await db
      .select()
      .from(schema.chiiTimeline)
      .where(op.eq(schema.chiiTimeline.id, id))
      .limit(1);
    expect(data).toBeDefined();
    expect(data?.cat).toBe(5);
    expect(data?.type).toBe(1);
    expect(data?.memo).toBe('^_^\n(bgm38)&gt;_&lt;');
  });
});
