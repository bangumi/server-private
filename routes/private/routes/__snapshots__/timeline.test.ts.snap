// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`timeline > should get all 1`] = `
Array [
  Object {
    "batch": false,
    "cat": 3,
    "createdAt": 1656458004,
    "id": 29006093,
    "memo": Object {
      "subject": Array [
        Object {
          "comment": "",
          "rate": 0,
          "subject": Object {
            "id": 262922,
            "images": Object {
              "common": "https://lain.bgm.tv/pic/cover/c/b2/f8/262922_u1OMP.jpg",
              "grid": "https://lain.bgm.tv/pic/cover/g/b2/f8/262922_u1OMP.jpg",
              "large": "https://lain.bgm.tv/pic/cover/l/b2/f8/262922_u1OMP.jpg",
              "medium": "https://lain.bgm.tv/pic/cover/m/b2/f8/262922_u1OMP.jpg",
              "small": "https://lain.bgm.tv/pic/cover/s/b2/f8/262922_u1OMP.jpg",
            },
            "info": "2018-10-03",
            "locked": false,
            "name": "Criminal Minds Season 14",
            "nameCN": "犯罪心理 第十四季",
            "nsfw": false,
            "type": 6,
          },
        },
      ],
    },
    "replies": 0,
    "source": 0,
    "type": 10,
    "uid": 287622,
    "user": Object {
      "avatar": Object {
        "large": "https://lain.bgm.tv/pic/user/l/icon.jpg",
        "medium": "https://lain.bgm.tv/pic/user/m/icon.jpg",
        "small": "https://lain.bgm.tv/pic/user/s/icon.jpg",
      },
      "id": 287622,
      "joinedAt": 0,
      "nickname": "nickname 287622",
      "sign": "sing 287622",
      "username": "287622",
    },
  },
  Object {
    "batch": false,
    "cat": 3,
    "createdAt": 1656457954,
    "id": 29006090,
    "memo": Object {
      "subject": Array [
        Object {
          "comment": "",
          "rate": 0,
          "subject": Object {
            "id": 214114,
            "images": Object {
              "common": "https://lain.bgm.tv/pic/cover/c/74/33/214114_6z9WZ.jpg",
              "grid": "https://lain.bgm.tv/pic/cover/g/74/33/214114_6z9WZ.jpg",
              "large": "https://lain.bgm.tv/pic/cover/l/74/33/214114_6z9WZ.jpg",
              "medium": "https://lain.bgm.tv/pic/cover/m/74/33/214114_6z9WZ.jpg",
              "small": "https://lain.bgm.tv/pic/cover/s/74/33/214114_6z9WZ.jpg",
            },
            "info": "2017年9月27日 / Glenn Kershaw / Jeff Davis, Breen Frazier / Joe Mantegna, Matthew Gray Gubler, A.J. Cook",
            "locked": false,
            "name": "Criminal Minds Season 13",
            "nameCN": "犯罪心理 第十三季",
            "nsfw": false,
            "type": 6,
          },
        },
      ],
    },
    "replies": 0,
    "source": 0,
    "type": 6,
    "uid": 287622,
    "user": Object {
      "avatar": Object {
        "large": "https://lain.bgm.tv/pic/user/l/icon.jpg",
        "medium": "https://lain.bgm.tv/pic/user/m/icon.jpg",
        "small": "https://lain.bgm.tv/pic/user/s/icon.jpg",
      },
      "id": 287622,
      "joinedAt": 0,
      "nickname": "nickname 287622",
      "sign": "sing 287622",
      "username": "287622",
    },
  },
  Object {
    "batch": false,
    "cat": 4,
    "createdAt": 1656457954,
    "id": 29006089,
    "memo": Object {
      "progress": Object {
        "batch": Object {
          "epsTotal": "22",
          "epsUpdate": 22,
          "subject": Object {
            "id": 214114,
            "images": Object {
              "common": "https://lain.bgm.tv/pic/cover/c/74/33/214114_6z9WZ.jpg",
              "grid": "https://lain.bgm.tv/pic/cover/g/74/33/214114_6z9WZ.jpg",
              "large": "https://lain.bgm.tv/pic/cover/l/74/33/214114_6z9WZ.jpg",
              "medium": "https://lain.bgm.tv/pic/cover/m/74/33/214114_6z9WZ.jpg",
              "small": "https://lain.bgm.tv/pic/cover/s/74/33/214114_6z9WZ.jpg",
            },
            "info": "2017年9月27日 / Glenn Kershaw / Jeff Davis, Breen Frazier / Joe Mantegna, Matthew Gray Gubler, A.J. Cook",
            "locked": false,
            "name": "Criminal Minds Season 13",
            "nameCN": "犯罪心理 第十三季",
            "nsfw": false,
            "type": 6,
          },
          "volsTotal": "??",
          "volsUpdate": 0,
        },
      },
    },
    "replies": 0,
    "source": 0,
    "type": 0,
    "uid": 287622,
    "user": Object {
      "avatar": Object {
        "large": "https://lain.bgm.tv/pic/user/l/icon.jpg",
        "medium": "https://lain.bgm.tv/pic/user/m/icon.jpg",
        "small": "https://lain.bgm.tv/pic/user/s/icon.jpg",
      },
      "id": 287622,
      "joinedAt": 0,
      "nickname": "nickname 287622",
      "sign": "sing 287622",
      "username": "287622",
    },
  },
  Object {
    "batch": false,
    "cat": 4,
    "createdAt": 1656375812,
    "id": 28993551,
    "memo": Object {
      "progress": Object {
        "single": Object {
          "episode": Object {
            "airdate": "2018-03-15",
            "comment": 0,
            "desc": "",
            "disc": 0,
            "duration": "",
            "id": 751421,
            "lock": false,
            "name": "Last Gasp",
            "nameCN": "",
            "sort": 16,
            "subjectID": 214114,
            "type": 0,
          },
          "subject": Object {
            "id": 214114,
            "images": Object {
              "common": "https://lain.bgm.tv/pic/cover/c/74/33/214114_6z9WZ.jpg",
              "grid": "https://lain.bgm.tv/pic/cover/g/74/33/214114_6z9WZ.jpg",
              "large": "https://lain.bgm.tv/pic/cover/l/74/33/214114_6z9WZ.jpg",
              "medium": "https://lain.bgm.tv/pic/cover/m/74/33/214114_6z9WZ.jpg",
              "small": "https://lain.bgm.tv/pic/cover/s/74/33/214114_6z9WZ.jpg",
            },
            "info": "2017年9月27日 / Glenn Kershaw / Jeff Davis, Breen Frazier / Joe Mantegna, Matthew Gray Gubler, A.J. Cook",
            "locked": false,
            "name": "Criminal Minds Season 13",
            "nameCN": "犯罪心理 第十三季",
            "nsfw": false,
            "type": 6,
          },
        },
      },
    },
    "replies": 0,
    "source": 0,
    "type": 2,
    "uid": 287622,
    "user": Object {
      "avatar": Object {
        "large": "https://lain.bgm.tv/pic/user/l/icon.jpg",
        "medium": "https://lain.bgm.tv/pic/user/m/icon.jpg",
        "small": "https://lain.bgm.tv/pic/user/s/icon.jpg",
      },
      "id": 287622,
      "joinedAt": 0,
      "nickname": "nickname 287622",
      "sign": "sing 287622",
      "username": "287622",
    },
  },
  Object {
    "batch": false,
    "cat": 4,
    "createdAt": 1656370302,
    "id": 28993219,
    "memo": Object {
      "progress": Object {
        "batch": Object {
          "epsTotal": "22",
          "epsUpdate": 15,
          "subject": Object {
            "id": 214114,
            "images": Object {
              "common": "https://lain.bgm.tv/pic/cover/c/74/33/214114_6z9WZ.jpg",
              "grid": "https://lain.bgm.tv/pic/cover/g/74/33/214114_6z9WZ.jpg",
              "large": "https://lain.bgm.tv/pic/cover/l/74/33/214114_6z9WZ.jpg",
              "medium": "https://lain.bgm.tv/pic/cover/m/74/33/214114_6z9WZ.jpg",
              "small": "https://lain.bgm.tv/pic/cover/s/74/33/214114_6z9WZ.jpg",
            },
            "info": "2017年9月27日 / Glenn Kershaw / Jeff Davis, Breen Frazier / Joe Mantegna, Matthew Gray Gubler, A.J. Cook",
            "locked": false,
            "name": "Criminal Minds Season 13",
            "nameCN": "犯罪心理 第十三季",
            "nsfw": false,
            "type": 6,
          },
          "volsTotal": "??",
          "volsUpdate": 0,
        },
      },
    },
    "replies": 0,
    "source": 0,
    "type": 0,
    "uid": 287622,
    "user": Object {
      "avatar": Object {
        "large": "https://lain.bgm.tv/pic/user/l/icon.jpg",
        "medium": "https://lain.bgm.tv/pic/user/m/icon.jpg",
        "small": "https://lain.bgm.tv/pic/user/s/icon.jpg",
      },
      "id": 287622,
      "joinedAt": 0,
      "nickname": "nickname 287622",
      "sign": "sing 287622",
      "username": "287622",
    },
  },
  Object {
    "batch": false,
    "cat": 3,
    "createdAt": 1656355851,
    "id": 28992589,
    "memo": Object {
      "subject": Array [
        Object {
          "comment": "",
          "rate": 0,
          "subject": Object {
            "id": 371546,
            "images": Object {
              "common": "https://lain.bgm.tv/pic/cover/c/c8/8a/371546_Df9ri.jpg",
              "grid": "https://lain.bgm.tv/pic/cover/g/c8/8a/371546_Df9ri.jpg",
              "large": "https://lain.bgm.tv/pic/cover/l/c8/8a/371546_Df9ri.jpg",
              "medium": "https://lain.bgm.tv/pic/cover/m/c8/8a/371546_Df9ri.jpg",
              "small": "https://lain.bgm.tv/pic/cover/s/c8/8a/371546_Df9ri.jpg",
            },
            "info": "2022年7月4日 / 仁昌寺義人 / 衣笠彰梧 （MF文庫J『ようこそ実力至上主義の教室へ』／KADOKAWA刊） / 森田和明",
            "locked": false,
            "name": "ようこそ実力至上主義の教室へ 2nd Season",
            "nameCN": "欢迎来到实力至上主义的教室 第二季",
            "nsfw": false,
            "type": 2,
          },
        },
      ],
    },
    "replies": 0,
    "source": 0,
    "type": 10,
    "uid": 287622,
    "user": Object {
      "avatar": Object {
        "large": "https://lain.bgm.tv/pic/user/l/icon.jpg",
        "medium": "https://lain.bgm.tv/pic/user/m/icon.jpg",
        "small": "https://lain.bgm.tv/pic/user/s/icon.jpg",
      },
      "id": 287622,
      "joinedAt": 0,
      "nickname": "nickname 287622",
      "sign": "sing 287622",
      "username": "287622",
    },
  },
  Object {
    "batch": false,
    "cat": 4,
    "createdAt": 1656285827,
    "id": 28979826,
    "memo": Object {
      "progress": Object {
        "single": Object {
          "episode": Object {
            "airdate": "2017-10-19",
            "comment": 0,
            "desc": "",
            "disc": 0,
            "duration": "",
            "id": 751409,
            "lock": false,
            "name": "Killer App",
            "nameCN": "",
            "sort": 4,
            "subjectID": 214114,
            "type": 0,
          },
          "subject": Object {
            "id": 214114,
            "images": Object {
              "common": "https://lain.bgm.tv/pic/cover/c/74/33/214114_6z9WZ.jpg",
              "grid": "https://lain.bgm.tv/pic/cover/g/74/33/214114_6z9WZ.jpg",
              "large": "https://lain.bgm.tv/pic/cover/l/74/33/214114_6z9WZ.jpg",
              "medium": "https://lain.bgm.tv/pic/cover/m/74/33/214114_6z9WZ.jpg",
              "small": "https://lain.bgm.tv/pic/cover/s/74/33/214114_6z9WZ.jpg",
            },
            "info": "2017年9月27日 / Glenn Kershaw / Jeff Davis, Breen Frazier / Joe Mantegna, Matthew Gray Gubler, A.J. Cook",
            "locked": false,
            "name": "Criminal Minds Season 13",
            "nameCN": "犯罪心理 第十三季",
            "nsfw": false,
            "type": 6,
          },
        },
      },
    },
    "replies": 0,
    "source": 0,
    "type": 2,
    "uid": 287622,
    "user": Object {
      "avatar": Object {
        "large": "https://lain.bgm.tv/pic/user/l/icon.jpg",
        "medium": "https://lain.bgm.tv/pic/user/m/icon.jpg",
        "small": "https://lain.bgm.tv/pic/user/s/icon.jpg",
      },
      "id": 287622,
      "joinedAt": 0,
      "nickname": "nickname 287622",
      "sign": "sing 287622",
      "username": "287622",
    },
  },
  Object {
    "batch": false,
    "cat": 3,
    "createdAt": 1656285346,
    "id": 28979800,
    "memo": Object {
      "subject": Array [
        Object {
          "comment": "",
          "rate": 0,
          "subject": Object {
            "id": 309445,
            "images": Object {
              "common": "https://lain.bgm.tv/pic/cover/c/87/e8/309445_aUj3o.jpg",
              "grid": "https://lain.bgm.tv/pic/cover/g/87/e8/309445_aUj3o.jpg",
              "large": "https://lain.bgm.tv/pic/cover/l/87/e8/309445_aUj3o.jpg",
              "medium": "https://lain.bgm.tv/pic/cover/m/87/e8/309445_aUj3o.jpg",
              "small": "https://lain.bgm.tv/pic/cover/s/87/e8/309445_aUj3o.jpg",
            },
            "info": "2002-01-00、2099-01-00 / 沙河 / sandbox",
            "locked": false,
            "name": "沙盒音乐条目1",
            "nameCN": "沙盒音乐条目",
            "nsfw": false,
            "type": 3,
          },
        },
      ],
    },
    "replies": 0,
    "source": 0,
    "type": 11,
    "uid": 287622,
    "user": Object {
      "avatar": Object {
        "large": "https://lain.bgm.tv/pic/user/l/icon.jpg",
        "medium": "https://lain.bgm.tv/pic/user/m/icon.jpg",
        "small": "https://lain.bgm.tv/pic/user/s/icon.jpg",
      },
      "id": 287622,
      "joinedAt": 0,
      "nickname": "nickname 287622",
      "sign": "sing 287622",
      "username": "287622",
    },
  },
  Object {
    "batch": false,
    "cat": 4,
    "createdAt": 1656281750,
    "id": 28979627,
    "memo": Object {
      "progress": Object {
        "batch": Object {
          "epsTotal": "22",
          "epsUpdate": 3,
          "subject": Object {
            "id": 214114,
            "images": Object {
              "common": "https://lain.bgm.tv/pic/cover/c/74/33/214114_6z9WZ.jpg",
              "grid": "https://lain.bgm.tv/pic/cover/g/74/33/214114_6z9WZ.jpg",
              "large": "https://lain.bgm.tv/pic/cover/l/74/33/214114_6z9WZ.jpg",
              "medium": "https://lain.bgm.tv/pic/cover/m/74/33/214114_6z9WZ.jpg",
              "small": "https://lain.bgm.tv/pic/cover/s/74/33/214114_6z9WZ.jpg",
            },
            "info": "2017年9月27日 / Glenn Kershaw / Jeff Davis, Breen Frazier / Joe Mantegna, Matthew Gray Gubler, A.J. Cook",
            "locked": false,
            "name": "Criminal Minds Season 13",
            "nameCN": "犯罪心理 第十三季",
            "nsfw": false,
            "type": 6,
          },
          "volsTotal": "??",
          "volsUpdate": 0,
        },
      },
    },
    "replies": 0,
    "source": 0,
    "type": 0,
    "uid": 287622,
    "user": Object {
      "avatar": Object {
        "large": "https://lain.bgm.tv/pic/user/l/icon.jpg",
        "medium": "https://lain.bgm.tv/pic/user/m/icon.jpg",
        "small": "https://lain.bgm.tv/pic/user/s/icon.jpg",
      },
      "id": 287622,
      "joinedAt": 0,
      "nickname": "nickname 287622",
      "sign": "sing 287622",
      "username": "287622",
    },
  },
  Object {
    "batch": false,
    "cat": 3,
    "createdAt": 1656276191,
    "id": 28979355,
    "memo": Object {
      "subject": Array [
        Object {
          "comment": "",
          "rate": 0,
          "subject": Object {
            "id": 214114,
            "images": Object {
              "common": "https://lain.bgm.tv/pic/cover/c/74/33/214114_6z9WZ.jpg",
              "grid": "https://lain.bgm.tv/pic/cover/g/74/33/214114_6z9WZ.jpg",
              "large": "https://lain.bgm.tv/pic/cover/l/74/33/214114_6z9WZ.jpg",
              "medium": "https://lain.bgm.tv/pic/cover/m/74/33/214114_6z9WZ.jpg",
              "small": "https://lain.bgm.tv/pic/cover/s/74/33/214114_6z9WZ.jpg",
            },
            "info": "2017年9月27日 / Glenn Kershaw / Jeff Davis, Breen Frazier / Joe Mantegna, Matthew Gray Gubler, A.J. Cook",
            "locked": false,
            "name": "Criminal Minds Season 13",
            "nameCN": "犯罪心理 第十三季",
            "nsfw": false,
            "type": 6,
          },
        },
      ],
    },
    "replies": 0,
    "source": 0,
    "type": 10,
    "uid": 287622,
    "user": Object {
      "avatar": Object {
        "large": "https://lain.bgm.tv/pic/user/l/icon.jpg",
        "medium": "https://lain.bgm.tv/pic/user/m/icon.jpg",
        "small": "https://lain.bgm.tv/pic/user/s/icon.jpg",
      },
      "id": 287622,
      "joinedAt": 0,
      "nickname": "nickname 287622",
      "sign": "sing 287622",
      "username": "287622",
    },
  },
  Object {
    "batch": false,
    "cat": 3,
    "createdAt": 1656276185,
    "id": 28979353,
    "memo": Object {
      "subject": Array [
        Object {
          "comment": "Hotch 的演员跟制片人起了争议被 CBS 停职，然后被解雇了... / 怎么一直在用旧凶手",
          "rate": 0,
          "subject": Object {
            "id": 194118,
            "images": Object {
              "common": "https://lain.bgm.tv/pic/cover/c/5f/65/194118_5c9H0.jpg",
              "grid": "https://lain.bgm.tv/pic/cover/g/5f/65/194118_5c9H0.jpg",
              "large": "https://lain.bgm.tv/pic/cover/l/5f/65/194118_5c9H0.jpg",
              "medium": "https://lain.bgm.tv/pic/cover/m/5f/65/194118_5c9H0.jpg",
              "small": "https://lain.bgm.tv/pic/cover/s/5f/65/194118_5c9H0.jpg",
            },
            "info": "2016-09-28",
            "locked": false,
            "name": "Criminal Minds Season 12",
            "nameCN": "犯罪心理 第12季",
            "nsfw": false,
            "type": 6,
          },
        },
      ],
    },
    "replies": 0,
    "source": 0,
    "type": 6,
    "uid": 287622,
    "user": Object {
      "avatar": Object {
        "large": "https://lain.bgm.tv/pic/user/l/icon.jpg",
        "medium": "https://lain.bgm.tv/pic/user/m/icon.jpg",
        "small": "https://lain.bgm.tv/pic/user/s/icon.jpg",
      },
      "id": 287622,
      "joinedAt": 0,
      "nickname": "nickname 287622",
      "sign": "sing 287622",
      "username": "287622",
    },
  },
  Object {
    "batch": false,
    "cat": 4,
    "createdAt": 1656276185,
    "id": 28979352,
    "memo": Object {
      "progress": Object {
        "batch": Object {
          "epsTotal": "22",
          "epsUpdate": 22,
          "subject": Object {
            "id": 194118,
            "images": Object {
              "common": "https://lain.bgm.tv/pic/cover/c/5f/65/194118_5c9H0.jpg",
              "grid": "https://lain.bgm.tv/pic/cover/g/5f/65/194118_5c9H0.jpg",
              "large": "https://lain.bgm.tv/pic/cover/l/5f/65/194118_5c9H0.jpg",
              "medium": "https://lain.bgm.tv/pic/cover/m/5f/65/194118_5c9H0.jpg",
              "small": "https://lain.bgm.tv/pic/cover/s/5f/65/194118_5c9H0.jpg",
            },
            "info": "2016-09-28",
            "locked": false,
            "name": "Criminal Minds Season 12",
            "nameCN": "犯罪心理 第12季",
            "nsfw": false,
            "type": 6,
          },
          "volsTotal": "??",
          "volsUpdate": 0,
        },
      },
    },
    "replies": 0,
    "source": 0,
    "type": 0,
    "uid": 287622,
    "user": Object {
      "avatar": Object {
        "large": "https://lain.bgm.tv/pic/user/l/icon.jpg",
        "medium": "https://lain.bgm.tv/pic/user/m/icon.jpg",
        "small": "https://lain.bgm.tv/pic/user/s/icon.jpg",
      },
      "id": 287622,
      "joinedAt": 0,
      "nickname": "nickname 287622",
      "sign": "sing 287622",
      "username": "287622",
    },
  },
  Object {
    "batch": false,
    "cat": 3,
    "createdAt": 1656257440,
    "id": 28976237,
    "memo": Object {
      "subject": Array [
        Object {
          "comment": "武器耐久是个坏文明",
          "rate": 0,
          "subject": Object {
            "id": 62229,
            "images": Object {
              "common": "https://lain.bgm.tv/pic/cover/c/60/d8/62229_5Yt0j.jpg",
              "grid": "https://lain.bgm.tv/pic/cover/g/60/d8/62229_5Yt0j.jpg",
              "large": "https://lain.bgm.tv/pic/cover/l/60/d8/62229_5Yt0j.jpg",
              "medium": "https://lain.bgm.tv/pic/cover/m/60/d8/62229_5Yt0j.jpg",
              "small": "https://lain.bgm.tv/pic/cover/s/60/d8/62229_5Yt0j.jpg",
            },
            "info": "2017年3月3日 / Nintendo Switch、Wii U / AAVG / 任天堂企画制作本部",
            "locked": false,
            "name": "ゼルダの伝説 ブレス オブ ザ ワイルド",
            "nameCN": "塞尔达传说 旷野之息",
            "nsfw": false,
            "type": 4,
          },
        },
      ],
    },
    "replies": 0,
    "source": 0,
    "type": 8,
    "uid": 287622,
    "user": Object {
      "avatar": Object {
        "large": "https://lain.bgm.tv/pic/user/l/icon.jpg",
        "medium": "https://lain.bgm.tv/pic/user/m/icon.jpg",
        "small": "https://lain.bgm.tv/pic/user/s/icon.jpg",
      },
      "id": 287622,
      "joinedAt": 0,
      "nickname": "nickname 287622",
      "sign": "sing 287622",
      "username": "287622",
    },
  },
  Object {
    "batch": false,
    "cat": 4,
    "createdAt": 1656257059,
    "id": 28976108,
    "memo": Object {
      "progress": Object {
        "single": Object {
          "episode": Object {
            "airdate": "2022-02-02",
            "comment": 0,
            "desc": "The 4th grade class is banned from wearing their pajamas to school on the most important day of the year.",
            "disc": 0,
            "duration": "22m",
            "id": 1080550,
            "lock": false,
            "name": "Pajama Day",
            "nameCN": "",
            "sort": 1,
            "subjectID": 366527,
            "type": 0,
          },
          "subject": Object {
            "id": 366527,
            "images": Object {
              "common": "https://lain.bgm.tv/pic/cover/c/0a/8f/366527_93yEY.jpg",
              "grid": "https://lain.bgm.tv/pic/cover/g/0a/8f/366527_93yEY.jpg",
              "large": "https://lain.bgm.tv/pic/cover/l/0a/8f/366527_93yEY.jpg",
              "medium": "https://lain.bgm.tv/pic/cover/m/0a/8f/366527_93yEY.jpg",
              "small": "https://lain.bgm.tv/pic/cover/s/0a/8f/366527_93yEY.jpg",
            },
            "info": "2022年2月2日 / Trey Parker",
            "locked": false,
            "name": "South Park Season 25",
            "nameCN": "南方公园 第25季",
            "nsfw": false,
            "type": 2,
          },
        },
      },
    },
    "replies": 0,
    "source": 0,
    "type": 2,
    "uid": 287622,
    "user": Object {
      "avatar": Object {
        "large": "https://lain.bgm.tv/pic/user/l/icon.jpg",
        "medium": "https://lain.bgm.tv/pic/user/m/icon.jpg",
        "small": "https://lain.bgm.tv/pic/user/s/icon.jpg",
      },
      "id": 287622,
      "joinedAt": 0,
      "nickname": "nickname 287622",
      "sign": "sing 287622",
      "username": "287622",
    },
  },
  Object {
    "batch": false,
    "cat": 3,
    "createdAt": 1656255672,
    "id": 28975618,
    "memo": Object {
      "subject": Array [
        Object {
          "comment": "",
          "rate": 0,
          "subject": Object {
            "id": 366527,
            "images": Object {
              "common": "https://lain.bgm.tv/pic/cover/c/0a/8f/366527_93yEY.jpg",
              "grid": "https://lain.bgm.tv/pic/cover/g/0a/8f/366527_93yEY.jpg",
              "large": "https://lain.bgm.tv/pic/cover/l/0a/8f/366527_93yEY.jpg",
              "medium": "https://lain.bgm.tv/pic/cover/m/0a/8f/366527_93yEY.jpg",
              "small": "https://lain.bgm.tv/pic/cover/s/0a/8f/366527_93yEY.jpg",
            },
            "info": "2022年2月2日 / Trey Parker",
            "locked": false,
            "name": "South Park Season 25",
            "nameCN": "南方公园 第25季",
            "nsfw": false,
            "type": 2,
          },
        },
      ],
    },
    "replies": 0,
    "source": 0,
    "type": 10,
    "uid": 287622,
    "user": Object {
      "avatar": Object {
        "large": "https://lain.bgm.tv/pic/user/l/icon.jpg",
        "medium": "https://lain.bgm.tv/pic/user/m/icon.jpg",
        "small": "https://lain.bgm.tv/pic/user/s/icon.jpg",
      },
      "id": 287622,
      "joinedAt": 0,
      "nickname": "nickname 287622",
      "sign": "sing 287622",
      "username": "287622",
    },
  },
  Object {
    "batch": false,
    "cat": 4,
    "createdAt": 1656092626,
    "id": 28941116,
    "memo": Object {
      "progress": Object {
        "batch": Object {
          "epsTotal": "11",
          "epsUpdate": 4,
          "subject": Object {
            "id": 347656,
            "images": Object {
              "common": "https://lain.bgm.tv/pic/cover/c/6d/d9/347656_5Sk6o.jpg",
              "grid": "https://lain.bgm.tv/pic/cover/g/6d/d9/347656_5Sk6o.jpg",
              "large": "https://lain.bgm.tv/pic/cover/l/6d/d9/347656_5Sk6o.jpg",
              "medium": "https://lain.bgm.tv/pic/cover/m/6d/d9/347656_5Sk6o.jpg",
              "small": "https://lain.bgm.tv/pic/cover/s/6d/d9/347656_5Sk6o.jpg",
            },
            "info": "2022-06-01",
            "locked": false,
            "name": "The Orville Season3",
            "nameCN": "奥维尔号 第三季",
            "nsfw": false,
            "type": 6,
          },
          "volsTotal": "??",
          "volsUpdate": 0,
        },
      },
    },
    "replies": 0,
    "source": 0,
    "type": 0,
    "uid": 287622,
    "user": Object {
      "avatar": Object {
        "large": "https://lain.bgm.tv/pic/user/l/icon.jpg",
        "medium": "https://lain.bgm.tv/pic/user/m/icon.jpg",
        "small": "https://lain.bgm.tv/pic/user/s/icon.jpg",
      },
      "id": 287622,
      "joinedAt": 0,
      "nickname": "nickname 287622",
      "sign": "sing 287622",
      "username": "287622",
    },
  },
  Object {
    "batch": false,
    "cat": 3,
    "createdAt": 1654396267,
    "id": 28685056,
    "memo": Object {
      "subject": Array [
        Object {
          "comment": "",
          "rate": 0,
          "subject": Object {
            "id": 240346,
            "images": Object {
              "common": "https://lain.bgm.tv/pic/cover/c/5a/b7/240346_LDL3v.jpg",
              "grid": "https://lain.bgm.tv/pic/cover/g/5a/b7/240346_LDL3v.jpg",
              "large": "https://lain.bgm.tv/pic/cover/l/5a/b7/240346_LDL3v.jpg",
              "medium": "https://lain.bgm.tv/pic/cover/m/5a/b7/240346_LDL3v.jpg",
              "small": "https://lain.bgm.tv/pic/cover/s/5a/b7/240346_LDL3v.jpg",
            },
            "info": "2018年10月5日 / 宅野誠起 / 金田陽介（講談社「週刊少年マガジン」連載中） / 森本由布希",
            "locked": false,
            "name": "寄宿学校のジュリエット",
            "nameCN": "寄宿学校的朱丽叶",
            "nsfw": false,
            "type": 2,
          },
        },
      ],
    },
    "replies": 0,
    "source": 0,
    "type": 2,
    "uid": 579876,
  },
  Object {
    "batch": false,
    "cat": 3,
    "createdAt": 1654396267,
    "id": 28685055,
    "memo": Object {
      "subject": Array [],
    },
    "replies": 0,
    "source": 1,
    "type": 4,
    "uid": 302300,
  },
  Object {
    "batch": false,
    "cat": 3,
    "createdAt": 1654396263,
    "id": 28685054,
    "memo": Object {
      "subject": Array [
        Object {
          "comment": "",
          "rate": 0,
          "subject": Object {
            "id": 386195,
            "images": Object {
              "common": "https://lain.bgm.tv/pic/cover/c/d0/d0/386195_E67b7.jpg",
              "grid": "https://lain.bgm.tv/pic/cover/g/d0/d0/386195_E67b7.jpg",
              "large": "https://lain.bgm.tv/pic/cover/l/d0/d0/386195_E67b7.jpg",
              "medium": "https://lain.bgm.tv/pic/cover/m/d0/d0/386195_E67b7.jpg",
              "small": "https://lain.bgm.tv/pic/cover/s/d0/d0/386195_E67b7.jpg",
            },
            "info": "2023年8月4日 / 石原立也 / 武田綾乃（宝島社文庫『響け! ユーフォニアム 北宇治高校吹奏楽部のホントの話』） / 池田晶子",
            "locked": false,
            "name": "特別編 響け！ユーフォニアム～アンサンブルコンテスト～",
            "nameCN": "特别篇 吹响！悠风号～合奏比赛～",
            "nsfw": false,
            "type": 2,
          },
        },
      ],
    },
    "replies": 0,
    "source": 0,
    "type": 2,
    "uid": 559989,
  },
  Object {
    "batch": false,
    "cat": 3,
    "createdAt": 1654396237,
    "id": 28685052,
    "memo": Object {
      "subject": Array [
        Object {
          "comment": "",
          "rate": 9,
          "subject": Object {
            "id": 1424,
            "images": Object {
              "common": "https://lain.bgm.tv/pic/cover/c/48/9d/1424_q8FMQ.jpg",
              "grid": "https://lain.bgm.tv/pic/cover/g/48/9d/1424_q8FMQ.jpg",
              "large": "https://lain.bgm.tv/pic/cover/l/48/9d/1424_q8FMQ.jpg",
              "medium": "https://lain.bgm.tv/pic/cover/m/48/9d/1424_q8FMQ.jpg",
              "small": "https://lain.bgm.tv/pic/cover/s/48/9d/1424_q8FMQ.jpg",
            },
            "info": "2009年4月2日 / 山田尚子 / かきふらい（芳文社「まんがタイムきらら」連載） / 堀口悠紀子",
            "locked": false,
            "name": "けいおん！",
            "nameCN": "轻音少女",
            "nsfw": false,
            "type": 2,
          },
        },
      ],
    },
    "replies": 0,
    "source": 1,
    "type": 2,
    "uid": 681772,
  },
]
`;

exports[`timeline > should get friends 1`] = `Array []`;
